app/models/blog.rb

```diff
@@ -10,11 +10,19 @@ class Blog < ApplicationRecord
   scope :published, -> { where('secret = FALSE') }
 
   scope :search, lambda { |term|
-    where('title LIKE ? OR content LIKE ?', "%#{term}%", "%#{term}%")
+    where('title LIKE :term OR content LIKE :term', term: "%#{term}%")
   }
 
   scope :default_order, -> { order(id: :desc) }
 
+  scope :viewable, lambda { |user|
+    if user
+      where('user_id = ?', user.id).or(published)
```

Comment:
```suggestion
      where('user_id = ?', user&.id).or(published)
```

のようにすると条件分岐がいらなくなるはずですー